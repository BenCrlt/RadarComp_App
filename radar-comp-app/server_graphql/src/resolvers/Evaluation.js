function eval_list_notes(parent, args, context) {
    return context.prisma.evaluation.findUnique({where : { eval_id: parent.eval_id }}).eval_list_notes();
}

module.exports = {
    eval_list_notes
}