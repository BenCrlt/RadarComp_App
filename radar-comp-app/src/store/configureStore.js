import {createStore} from 'redux';
import addItemValue from './reducers/evalReducer';

export default createStore(addItemValue);