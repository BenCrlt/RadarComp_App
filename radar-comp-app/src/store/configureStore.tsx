import {createStore, combineReducers, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

//import evalReducer from './evaluation/reducer';
import commonReducer from './common/reducer';

const persistConfig = {
    key: 'root',
    storage
}

const reducer = combineReducers({
    //eval: evalReducer,
    common: persistReducer(persistConfig, commonReducer), 
})

var store = createStore(reducer, applyMiddleware(thunk));

export const persistor = persistStore(store);

export default store