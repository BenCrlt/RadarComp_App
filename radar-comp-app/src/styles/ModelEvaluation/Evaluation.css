.rca-eval {
    display: grid;
    grid-template-columns: 5% [start-container] 600px [finish-container] auto;
    grid-template-rows: 30px [list-skills-start] auto [list-skills-end] 30px [start-buttons];
} 

html, body {
    overflow: auto;
}

.rca-eval-skill-list-container {
    grid-column: start-container / finish-container;
    grid-row: list-skills-start / list-skills-end;
}

.rca-eval-create-eval-container {
    grid-column: start-container / finish-container;
    grid-row: start-buttons;
  
}   