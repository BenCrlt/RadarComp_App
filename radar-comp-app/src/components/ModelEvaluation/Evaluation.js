import '../../styles/ModelEvaluation/Evaluation.css'
import {listEval} from '../../datas/formEval.js'
import Skill from './Skill';

function Evaluation({user}) {
    const rangeScale = 5;
    return (
        <div className="rca-eval">
            <h1 className="border-bottom">Evaluation {getDateString()}</h1>
                <div className="rca-eval-skill-list">
                    {listEval.map((skill) => (
                        <Skill skill={skill} key={skill.id} rangeScale={rangeScale}/>
                    ))}
                </div>
                <div className="rca-eval-skill-button">
                    <button type="button" className="btn btn-primary btn-lg">Envoyer</button>   
                </div>
        </div>
    )
}

function getDateString() {
    const date = new Date(),
        day = date.getDay() >= 10 ? date.getDay() : "0" + date.getDay(),
        month = date.getMonth() >= 10 ? (date.getMonth() + 1) : "0" + (date.getMonth()+1),
        year = date.getFullYear();
    return day + "/" + month + "/" + year;
}
export default Evaluation;