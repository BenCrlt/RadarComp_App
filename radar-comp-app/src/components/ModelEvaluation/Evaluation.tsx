import '../../styles/ModelEvaluation/Evaluation.css'
import { useState, useEffect } from 'react';
import { connect, ConnectedProps } from 'react-redux';
import axios from 'axios';
import Skill from './Skill';

type PropsFromRedux = ConnectedProps<typeof connector>;

function Evaluation({listValueItems} : PropsFromRedux) {
    const [listSkills, setListSkills] = useState<SkillType[]>([]);
    useEffect(() => {
        axios.get('http://localhost:3000/api/list_skills')
        .then(res => {
            setListSkills(res.data);
        });
    }, [])
    const rangeScale = 5;
    const date = new Date().toLocaleDateString();
    return (
        <div className="rca-eval">
            <h1 className="border-bottom">Evaluation {date}</h1>
                <div className="rca-eval-skill-list">
                    {listSkills.map((skill) => (
                        <Skill skill={skill} key={skill.skill_id} rangeScale={rangeScale}/>
                    ))}
                </div>
                <div className="rca-eval-skill-button">
                    <button type="button" className="btn btn-primary btn-lg" onClick={() => SendEvaluation(listValueItems)}>Envoyer</button>
                </div>
        </div>
    )
}

function SendEvaluation(listItems : ListItemsType[]) {
    console.log(listItems);
}

const mapStateToProps = (state : StateType) => {
    return {
        listValueItems: state.listValueItems
    }
}

const connector = connect(mapStateToProps);

export default connector(Evaluation);