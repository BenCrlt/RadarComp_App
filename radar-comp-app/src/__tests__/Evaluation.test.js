import React from 'react';
import Evaluation from '../components/ModelEvaluation/Evaluation'
import {create} from 'react-test-renderer'
import { Provider } from 'react-redux';
import store from '../store/configureStore'
import { ApolloProvider, ApolloClient, InMemoryCache} from '@apollo/client';

describe('Evaluation', () => {
    let client;
    beforeEach(() => {
        client = new ApolloClient({
            uri: 'http://localhost:4000/',
            cache: new InMemoryCache()
        });
    })
    test('Displays the information when the query is loaded', () => {
        const evaluationComponent = create(
            <ApolloProvider client={client}>
                <Provider store={store}>
                    <Evaluation/>
                </Provider>
            </ApolloProvider>
        );
        let tree = evaluationComponent.toJSON();
        expect(tree).toMatchSnapshot();
    }) 
})

