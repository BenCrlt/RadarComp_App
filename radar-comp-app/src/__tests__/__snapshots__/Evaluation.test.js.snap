// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluation Displays the information when the query is loaded 1`] = `
<div
  className="rca-eval"
>
  <h1
    className="border-bottom"
  >
    Evaluation 
    01/10/2021
  </h1>
  <div
    className="rca-eval-skill-list"
  />
  <div
    className="rca-eval-skill-button"
  >
    <button
      className="btn btn-primary btn-lg"
      onClick={[Function]}
      type="button"
    >
      Envoyer
    </button>
  </div>
</div>
`;

exports[`Evaluation Displays the information when the query is loaded 2`] = `
<div
  className="rca-eval"
>
  <h1
    className="border-bottom"
  >
    Evaluation 
    01/10/2021
  </h1>
  <div
    className="rca-eval-skill-list"
  />
  <div
    className="rca-eval-skill-button"
  >
    <button
      className="btn btn-primary btn-lg"
      onClick={[Function]}
      type="button"
    >
      Envoyer
    </button>
  </div>
</div>
`;
